streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -623,6 +623,21 @@ private void addToTasksToClose(final Map<TaskId, CompletableFuture<StateUpdater.
         });
     }
 
+    private void iterateAndActOnRemovedTask(final Map<TaskId, CompletableFuture<StateUpdater.RemovedTaskResult>> futures,
```

Comment:
Are we going to reuse this method? Otherwise, I find this a bit too abstract, and I'd just inline it.

Reply:
Yes, we are going to re-use it. Stay tuned!

Reply:
OK then. Let's continue to have it as a method.